language/go/resolve.go

```diff
@@ -288,11 +290,20 @@ func resolveExternal(moduleMode bool, rc *repo.RemoteCache, imp string) (label.L
 		pkg = pathtools.TrimPrefix(impWithoutSemver, prefix)
 	}
 
-	return label.New(repo, pkg, defaultLibName), nil
+	nc := gc.goNamingConvention
```

Comment:
Add a comment:

```
// Determine what naming convention is used by the go_repository rule.
// If none is specified with "build_naming_convention", the either naming convention
// can be used, so we'll default to whatever's used in the current workspace to avoid churn.
```